lib/spack/spack/stage.py

```diff
@@ -725,6 +738,148 @@ def _add_to_root_stage(self):
                     install(src, destination_path)
 
 
+class CMakeBuildStage(Stage):
+    """Interface abstracting a CMake build tree at a location outside
+    of a Spack stage directory but still managed by Spack.
+
+    CMake's build tree can be located arbitrarily on a filesystem independent
+    from a source directory. This class relocates that build directory out of the
+    stage and to a path under a Users home directory or to a location of a users chosing set
+    either by config or the command line via the --cmake-build-dir cl argument
+    to the install command.
+    Interactions with the stage expecting an in stage build tree will work as normal
+    as this class serves to obfucscate the external stage and allow for all stage behavior
+    to perform as normal.
+    The external build tree is given the same lifespan as its stage dir, is spun up when the stage
+    spins up, and destroyed when the stage is destroyed. After the build and installation are done,
+    the build tree is first relocated to what would be its proper place in the stage.
+    and a symlink is placed in place of the build tree pointing at the external directory
+
+    Note: This is not, nor should it be, used on *nix platforms and is intended as a solution to
+    reduce file path lengths on Windows during compilation/linking. This class should be removed
+    when MSVC fully supports the LongPath feature on Windows.
+    """
+
+
+    def __init__(self, hash, fetcher, **kwargs):
+        # Users can override external cmake build dir, default is %USERPROFILE%
+        # overrides can come from command line or config, command line will override all
+        super(CMakeBuildStage, self).__init__(fetcher, **kwargs)
+        self._hash = hash
+        self._remote_stage = None
```

Comment:
Can you rename this to `_remote_build_dir`? I want to avoid using the token "stage" since that might be taken to imply that it is a `Stage` object when it is in fact a string (or maybe a `Path`, but something with less functionality than a `Stage` in any case).

Side note: I don't see anything setting this variable (e.g. `_remote_stage = ...`, other than when it is set to `None`)

Reply:
This variable is set in `create`